src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaCoverageProvider.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015 Uber
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package com.uber.jenkins.phabricator.coverage;
+
+import hudson.model.AbstractBuild;
+import hudson.plugins.cobertura.CoberturaBuildAction;
+import hudson.plugins.cobertura.Ratio;
+import hudson.plugins.cobertura.targets.CoverageMetric;
+import hudson.plugins.cobertura.targets.CoverageResult;
+
+/**
+ * Provide Cobertura coverage data
+ */
+@SuppressWarnings("unused")
+public class CoberturaCoverageProvider extends CoverageProvider {
+    @Override
+    public boolean hasCoverage() {
+        CoverageResult result = getCoverageResult();
+        return result != null && result.getCoverage(CoverageMetric.LINE) != null;
+    }
+
+    @Override
+    public CodeCoverageMetrics getMetrics() {
+        CoverageResult result = getCoverageResult();
+        if (result == null) {
```

Comment:
1 minor thing. Since you're doing this check inside the convert method, this is redundant. Another option is to move the entire convert method inside getMetrics(). Really up to you, ok with either.